src/match/controllers/update_match_controller.py

```diff
@@ -0,0 +1,121 @@
+import datetime
+import json
+
+from api.utils import failure_response
+from api.utils import success_response
+from location.models import Location
+from match import match_status
+from match.models import Match
+from rest_framework import status
+
+
+class UpdateMatchController:
+    def __init__(self, match_id, user, data, serializer):
+        self._match_id = match_id
+        self._user = user
+        self._data = data
+        self._serializer = serializer
+
+    def process(self):
+        # First, find the match and any fields to be updated
+        self._match = Match.objects.filter(id=self._match_id)
+        if not self._match:
+            return failure_response("Match does not exist.")
+        self._match = self._match[0]
+        proposed_meeting_times = self._data.get("proposed_meeting_times")
+        proposed_locations = self._data.get("proposed_locations")
+        meeting_location_id = self._data.get("meeting_location")
+        meeting_time = self._data.get("meeting_time")
+
+        # Next, modify the attributes that may have changed:
+        # Proposed Meeting Times
+        self._modify_attribute("proposed_meeting_times", proposed_meeting_times)
+        # Meeting Time
+        if meeting_time is not None:
+            self._update_meeting_time(meeting_time)
+        # Proposed Locations
+        if proposed_locations is not None:
```

Comment:
any reason why you go through the effort to check if a Location is already in the previous match object before adding and removing any locations whose ids were not in the proposed_location instead of just building up a new query set to assign `_match.proposed_locations` to?